crates/volta-core/src/tool/package/mod.rs

```diff
@@ -59,15 +59,15 @@ impl Package {
         Package { name, details }
     }
 
-    fn fetch_internal(&self) -> Fallible<()> {
+    fn fetch_internal(&self, mut session: &mut Session) -> Fallible<()> {
         // ISSUE(#288) - Once we have a valid Collection, we can check that in the same way as node/yarn
         // Until then, we use the existence of the image directory as the indicator that the package is
         // already fetched
         if path::package_image_dir(&self.name, &self.details.version.to_string())?.exists() {
             debug_already_fetched(self);
             Ok(())
         } else {
-            fetch::fetch(&self.name, &self.details)
+            fetch::fetch(&self.name, &self.details, &mut session)
```

Comment:
And here, since the session is already a mutable reference, we can pass it along directly:

```suggestion
            fetch::fetch(&self.name, &self.details, session)
```